apps/user-office-frontend/src/components/questionary/questionaryComponents/GenericTemplate/GenericTemplateSelectModalOnCopy.tsx

```diff
@@ -148,107 +150,106 @@ const GenericTemplateSelectModalOnCopy = ({
 
       <Grid container spacing={3}>
         <Grid item xs={12}>
-          {genericTemplates && (
-            <div className={classes.selectorContainer}>
-              <Autocomplete
-                id="generic-template-proposal-select"
-                aria-labelledby="generic-template-proposal-select-label"
-                fullWidth={true}
-                noOptionsText={capitalize(
-                  `No proposal(s) with matching ${question || 'answers(s)'}`
-                )}
-                disableClearable
-                onChange={(_event, selectedValue) => {
-                  setGenericTemplateAnswer([]);
-                  setSelectedProposalPk(selectedValue.proposalPk);
+          <div className={classes.selectorContainer}>
+            <Autocomplete
+              id="generic-template-proposal-select"
+              aria-labelledby="generic-template-proposal-select-label"
+              fullWidth={true}
+              noOptionsText={capitalize(
+                `No proposal(s) with matching ${question || 'answers(s)'}`
+              )}
+              disableClearable
+              onChange={(_event, selectedValue) => {
+                setGenericTemplateAnswer([]);
+                setSelectedProposalPk(selectedValue.proposalPk);
+              }}
+              getOptionLabel={(option) => option.proposal.title}
+              isOptionEqualToValue={(option, value) =>
+                option.proposalPk === value.proposalPk
+              }
+              options={genericTemplates.filter(
+                (template, index) =>
+                  genericTemplates.findIndex(
+                    (value) => value.proposalPk == template.proposalPk
+                  ) === index
+              )}
+              renderInput={(params) => (
+                <TextField
+                  {...params}
+                  key={params.id}
+                  data-cy="genericTemplateProposalTitle"
+                  label="Proposal title"
+                  placeholder="Select proposal title"
+                />
+              )}
+              renderOption={(props, option) => {
+                return (
+                  <li {...props} key={option.proposalPk}>
+                    {option.proposal.title}
+                  </li>
+                );
+              }}
+              loading={isExecutingCall}
+              loadingText={'Please wait loading proposal title(s)'}
```

Comment:
I think here a comma is needed 

"Please wait, loading proposal title(s)"

Or maybe just "Loading..."

Reply:
The default  if loading so have removed customisation